app/src/main/java/org/intelehealth/app/activities/notification/view/NotificationActivity.kt

```diff
@@ -127,26 +130,41 @@ class NotificationActivity : BaseActivity(), ClearNotificationListener {
                         getString(R.string.this_visit_is_completed)
                     )
                 } else {
-                    val intent = Intent(this@NotificationActivity, PrescriptionActivity::class.java)
-                    intent.putExtra("patientname", "$first_name $last_name")
-                    intent.putExtra("patientUuid", patientuuid)
-                    intent.putExtra("patient_photo", patient_photo)
-                    intent.putExtra("visit_ID", visitUUID)
-                    intent.putExtra("visit_startDate", visit_startDate)
-                    intent.putExtra("gender", gender)
-                    val vitalsUUID = EncounterDAO.fetchEncounterUuidForEncounterVitals(visitUUID)
-                    val adultInitialUUID =
-                        EncounterDAO.fetchEncounterUuidForEncounterAdultInitials(visitUUID)
-                    intent.putExtra("encounterUuidVitals", vitalsUUID)
-                    intent.putExtra("encounterUuidAdultIntial", adultInitialUUID)
-                    intent.putExtra(
-                        "age",
-                        DateAndTimeUtils.getAge_FollowUp(date_of_birth, this@NotificationActivity)
-                    )
-                    intent.putExtra("tag", "VisitDetailsActivity")
-                    intent.putExtra("followupDate", followupDate)
-                    intent.putExtra("openmrsID", openmrsID)
-                    startActivity(intent)
+                    if(notificationModel.notification_type == NotificationDbConstants.FOLLOW_UP_NOTIFICATION){
+                        val intent = Intent(IntelehealthApplication.getAppContext(), VisitSummaryActivity_New::class.java).apply {
+                            putExtra("patientUuid", patientuuid)
```

Comment:
Instead of sharing data in multiple params, we can create param model class to set as put extra, key should be a constant variable instead of hardcoded.